### ISSUE TITLE
SIGSEGV segfault

### ISSUE THREAD

**Issue by bitwave** [kleiner-brauhelfer/kleiner-brauhelfer-2#89]
When I select a Sud in the overview, kleiner-brauhelfer crashes with a segfault.

**Comment by bitwave** [comment]
Qtcreator says it crashes in `tabsudauswahl_sudinfo.cpp:301` in statement
```cpp
map.insert("Einheit", TabRohstoffe::Einheiten[static_cast<int>(eintrag.Einheit)]);
```
maybe the calculated address is incorrect.

**Comment by bitwave** [comment]
The eintrag.Einheit value is unitialized (very large, negative value). Therefore the `Einheiten[-4334143]` is out of bounds.

**Comment by bitwave** [comment]
I solved the issue by changing the Einheit value in the sqlitedb from -4729231 to 0. But maybe the database schema upgrade code is broken...

**Comment by BourgeoisLab** [comment]
Is it a Sud that you created with kbh version 2 or 1?

**Comment by realholgi** [comment]
There is sometimes a value of 512 instead of 1 in Einheit. You have to fix this manually in the Database to avoid the crash.

**Comment by BourgeoisLab** [comment]
The question is where do these values come from. I will implement a check in the "Datenbank reinigen" procedure. But i will probably let the program crash instead of using a "random" Einheit of 0 as a fallback.

**Comment by realholgi** [comment]
> The question is where do these values come from. 

In my case, it came over from my KBH1 Database.

<img width="1039" alt="shot" src="https://user-images.githubusercontent.com/980419/79121070-bac00680-7d94-11ea-93f2-c8c2ddcc5743.png">

**Comment by BourgeoisLab** [comment]
By the way here the definition
`    enum class Einheit
    {
        Kg = 0,
        g = 1,
        mg = 2,
        Stk = 3,
        l,
        ml
    };`

**Comment by bitwave** [comment]
It came from kleiner brauhelfer 1 originally, but i used it with 2.x.x branch before. The database cleanup tool was used before.

**Comment by BourgeoisLab** [comment]
OK. Important for me is to know, that these strange numbers don't come from v2.
The cleanup doesn't chek this at the moment, but I will implement something for the next version.

**Comment by BourgeoisLab** [comment]
How to manually fix:
https://hobbybrauer.de/forum/viewtopic.php?p=376153#p376153

**Comment by BourgeoisLab** [comment]
Still crashing, but database cleaner implemented in 21aee9546252c842b8de2fc3bdc5604082f38d9d.

**Comment by LondonRain** [comment]
> 
> 
> Still crashing, but database cleaner implemented in [21aee95](https://github.com/kleiner-brauhelfer/kleiner-brauhelfer-2/commit/21aee9546252c842b8de2fc3bdc5604082f38d9d).

Ich bitte das noch mal zu überdenken. Siehe:
https://hobbybrauer.de/forum/viewtopic.php?f=3&t=21510&p=378378#p378378
